polygon/heimdall/entity_store.go

```diff
@@ -36,25 +46,53 @@ func newEntityStore(
 		makeEntity:      makeEntity,
 		getLastEntityId: getLastEntityId,
 		loadEntityBytes: loadEntityBytes,
+
+		blockNumToIdIndex: blockNumToIdIndex,
 	}
 }
 
+func (s *entityStoreImpl) Prepare() error {
```

Comment:
im only seeing this now but I was under the impression we discussed not to use `Impl` suffixes - can rename the struct to just `store` - or remove the interface and just use a struct if it is not needed for unit tests?

Reply:
The current convention is:
```
MyService - interface
myService - implementation
```
but what if I want the interface be private?
We need to establish a convention for this case.

In this case I agree to remove it, and convert everything to a pointer to a struct, but I feel that I lose some value if I'm doing that. `entityStore` is used in 14 places, and I prefer to keep the door open for mocking and reuse. Also given my background I like to have an interface on top that summarizes what's implemented, that's more of a personal preference.